core/src/main/java/org/mobilitydata/gtfsvalidator/notice/Notice.java

```diff
@@ -40,11 +42,18 @@ public SeverityLevel getSeverityLevel() {
   }
 
   /**
-   * Returns a descriptive type-specific name for this notice.
+   * Returns a descriptive type-specific name for this notice based on the class simple name.
    *
    * @return notice code, e.g., "foreign_key_error".
    */
-  public abstract String getCode();
+  public String getCode() {
+    String toReturn =
```

Comment:
A more efficient implementation would be:

```java
String code = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, getClass().getSimpleName());
if (code.endsWith(NOTICE_SUBSTRING)) {
    return code.substring(0, code .length() - NOTICE_SUBSTRING.length());
}
return code;
```

...so the VM doesn't have to search for the string.